### DISCUSSION TITLE
How to integrate custom rp2040 version of openocd (+configs) into 'scons program' command?

### DISCUSSION THREAD

**Discussion by cocasema** [modm-io/modm#845] [Q&A] [answered]
Raspberry Pi Pico provides its own version of openocd which can be used with picoprobe (additional rp2040 board) to flash and debug the board. It would be extremely convenient to have it working via `scons program`  similarly to stm32+stlink.

Is there a way to customize scons files to support this?

`/path/to/pico/openocd -f interface/picoprobe.cfg -f target/rp2040.cfg -s tcl -c "program /path/to/program.elf verify reset exit"`

**Comment by salkinium** [comment, marked as answer]
Yeah, I saw that too and was hoping they would merge it back into upstream by the time I got around to dealing with it.
You would need to adapt the openocd command here and make the binary an argument: https://github.com/modm-io/modm/blob/develop/tools/modm_tools/openocd.py#L81
Then in the SCons wrapper you query it from the command line via `ARGUMENTS.get("path", "openocd")` (usage: `scons program path=/path/to/pico/openocd`): https://github.com/modm-io/modm/blob/develop/tools/build_script_generator/scons/site_tools/openocd.py#L47

It's not super elegant, but finding the path automatically is probably not doable generically due to installation location differences per OS.

**Reply by salkinium** [reply]
This has been fixed in https://github.com/modm-io/modm/pull/846 via an environmental variable.

```sh
export MODM_OPENOCD_BINARY=/path/to/openocd
```
